### PR TITLE
[OKR][BZ-2014304]:  Support --limit and --tags option RHCS4

### PR DESCRIPTION
**Pull request by Ameenasuhani**
Signed-off-by: ameenasuhani <ameenasuhani@gmail.com>

Description: Support --limit and --tags option for BZ-2014304
Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-R3ZOF8/

### CODE REVIEW

File: tests/ceph_ansible/test_ansible_upgrade.py [hunk 1]
```diff
@@ -146,6 +148,15 @@ def run(ceph_cluster, **kw):
     if config.get("ansi_cli_args"):
         cmd += config_dict_to_string(config["ansi_cli_args"])
 
+    if limit_node:
+        short_name_list = [
```
Comment (sunilkumarn417) [inline tests/ceph_ansible/test_ansible_upgrade.py:152]: I would suggest to use below definitions to get node(s). so we could re-use common code.
https://github.com/red-hat-storage/cephci/blob/master/ceph/utils.py#L978

File: suites/nautilus/upgrades/tier-2_upgrade_with-tags.yaml [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+      sudo: True
+    desc: Check for ceph mon dump debug info
+
+- test:
+    name: check-osd-dump
+    module: exec.py
+    config:
+      cmd: ceph osd dump
+      sudo: True
+    desc: Check for ceph osd dump debug info
```
Comment (sunilkumarn417) [inline suites/nautilus/upgrades/tier-2_upgrade_with-tags.yaml:527]: please have a look if we could club these commands and have it in one test case.

File: tests/ceph_ansible/test_ansible_upgrade.py (hunk 1 above)
Comment (sunilkumarn417) [reply tests/ceph_ansible/test_ansible_upgrade.py:152]: these methods would be helpful even if we have multiple node names.

**Review by sunilkumarn417 — COMMENTED** [review]
Hi @Ameenasuhani ,

Thanks for working on this improvements.

IMO, this would fail at version comparison if we use containerized deployment.
please check if skip-version-check would skip this too.

**Comment by Ameenasuhani** [comment]
@sunilkumarn417 as far as I know this is a customer scenario and will not be available to others as discussed earlier,
so are `containerized deployments` into consideration?

**Comment by sunilkumarn417** [comment]
> @sunilkumarn417 as far as I know this is a customer scenario and will not be available to others as discussed earlier, so are `containerized deployments` into consideration?

It would be good to support both.

**Review by sunilkumarn417 — COMMENTED** [review]
Hi @Ameenasuhani ,

can you please create respective pipeline shell script for this test suite under 4x.

**Review by sunilkumarn417 — APPROVED** [review]
LGTM

**Pull request merged**